ecosystem/typescript/sdk/src/faucet_client.test.ts

```diff
@@ -0,0 +1,59 @@
+import { AptosClient } from "./aptos_client";
+import FaucetClient from "./faucet_client";
+import AptosAccount from "./aptos_account";
+import Types from "./types";
+import { UserTransaction } from "./api/data-contracts";
+
+const nodeUrl = "https://fullnode.devnet.aptoslabs.com";
+const faucetUrl = "https://faucet.devnet.aptoslabs.com";
+
+
+test("full tutorial faucet flow", async () => {
+  const client = new AptosClient(nodeUrl);
+  const faucetClient = new FaucetClient(nodeUrl, faucetUrl);
+
+  const account1 = new AptosAccount();
+  const txns = await faucetClient.fundAccount(account1.address(), 5000);
+  const tx1 = await client.getTransaction(txns[1]);
+  expect(tx1.type).toBe("user_transaction");
+  let resources = await client.getAccountResources(account1.address());
+  let accountResource = resources.find((r) => r.type === "0x1::TestCoin::Balance");
+  expect((accountResource.data as { "coin": { "value": string } }).coin.value).toBe("5000");
+
+  const account2 = new AptosAccount();
+  await faucetClient.fundAccount(account2.address(), 0);
+  resources = await client.getAccountResources(account2.address());
+  accountResource = resources.find((r) => r.type === "0x1::TestCoin::Balance");
+  expect((accountResource.data as { "coin": { "value": string } }).coin.value).toBe("0");
+
+  const payload: Types.TransactionPayload = {
+    type: "script_function_payload",
+    function: "0x1::TestCoin::transfer",
+    type_arguments: [],
+    arguments: [
+      account2.address().hex(),
+      "717",
+    ]
+  };
+  const txnRequest = await client.generateTransaction(account1.address(), payload);
+  const signedTxn = await client.signTransaction(account1, txnRequest);
+  const transactionRes = await client.submitTransaction(account1, signedTxn);
+  await client.waitForTransaction(transactionRes.hash);
+
+  resources = await client.getAccountResources(account2.address());
+  accountResource = resources.find((r) => r.type === "0x1::TestCoin::Balance");
+  expect((accountResource.data as { "coin": { "value": string } }).coin.value).toBe("717");
+
+  const res = await client.getAccountTransactions(account1.address(), { start: 0 });
+  const tx = res.find((e) => e.type === "user_transaction") as UserTransaction;
+  expect(tx.sender).toBe(account1.address().hex());
+
+  const events = await client.getEventsByEventHandle(
+    tx.sender,
+    "0x1::TestCoin::TransferEvents",
+    "sent_events");
```

Comment:
this is a bit weird syntax from what I've seen

any reason for not doing:
```
function_name(
  one_element,
  another_element,
)
```

I think that makes it slightly easier to read and if we end up changing the set of parameters it maintains appropriate "git blame"